library/src/main/java/com/airbnb/android/airmapview/RuntimePermissionUtils.java

```diff
@@ -0,0 +1,99 @@
+package com.airbnb.android.airmapview;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.Fragment;
+
+import static android.support.v4.content.PermissionChecker.checkSelfPermission;
+
+/**
+ * Utility class that handles Runtime location permissions for Android M
+ */
+public final class RuntimePermissionUtils {
+
+  private RuntimePermissionUtils() {
+  }
+
+  private static final byte LOCATION_PERMISSION_REQUEST_CODE = 1;
+  private static final String[] LOCATION_PERMISSIONS =
+          new String[]{"android.permission.ACCESS_FINE_LOCATION", "android.permission.ACCESS_COARSE_LOCATION"};
+
+  /**
+   * Checks all given permissions have been granted.
+   *
+   * @param grantResults results
+   * @return returns true if all permissions have been granted.
+   */
+  public static boolean verifyPermissions(int... grantResults) {
+    for (int result : grantResults) {
+      if (result != PackageManager.PERMISSION_GRANTED) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns true if the Activity or Fragment has access to all given permissions.
```

Comment:
It's true I can update that. Although this utils class is expecting an activity or fragment  since only activity and fragment have the onRequestPermissionsResult callback and usually it is the best practice to bound the runtime permission check within the activity or fragment life cycle.